src/librbd/Utils.h

```diff
@@ -152,6 +203,33 @@ Context *create_context_callback(T *obj) {
   return new detail::C_StateCallbackAdapter<T, MF, destroy>(obj);
 }
 
+//for reference counting objects
+template <typename T, void(T::*MF)(int) = &T::complete>
+Context *create_context_callback(T *obj, RefCountedPtr refptr) {
+  return new detail::C_RefCallbackAdapter<T, MF>(obj, refptr);
+}
+
+template <typename T, Context*(T::*MF)(int*), bool destroy=true>
```

Comment:
`destroy` param unneeded -- drop it an explicitly pass `true` below

Reply:
Done